src/AppInstallerRepositoryCore/Microsoft/Schema/1_1/Interface_1_1.cpp

```diff
@@ -0,0 +1,219 @@
+
+        // Update new 1:N tables as necessary
+        indexModified = PackageFamilyNameTable::UpdateIfNeededByManifestId(connection, GetPackageFamilyNames(manifest), manifestId) || indexModified;
+        indexModified = ProductCodeTable::UpdateIfNeededByManifestId(connection, GetProductCodes(manifest), manifestId) || indexModified;
+
+        savepoint.Commit();
+
+        return { indexModified, manifestId };
+    }
+
+    SQLite::rowid_t Interface::RemoveManifest(SQLite::Connection& connection, const Manifest::Manifest& manifest, const std::filesystem::path& relativePath)
+    {
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "removemanifest_v1_1");
+
+        SQLite::rowid_t manifestId = V1_0::Interface::RemoveManifest(connection, manifest, relativePath);
+
+        // Remove all of the new 1:N data that is no longer referenced.
+        PackageFamilyNameTable::DeleteIfNotNeededByManifestId(connection, manifestId);
+        ProductCodeTable::DeleteIfNotNeededByManifestId(connection, manifestId);
+
+        savepoint.Commit();
+
+        return manifestId;
+    }
+
+    void Interface::PrepareForPackaging(SQLite::Connection& connection)
+    {
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "prepareforpackaging_v1_1");
+
+        V1_0::IdTable::PrepareForPackaging(connection);
+        V1_0::NameTable::PrepareForPackaging(connection);
+        V1_0::MonikerTable::PrepareForPackaging(connection);
+        V1_0::VersionTable::PrepareForPackaging(connection);
+        V1_0::ChannelTable::PrepareForPackaging(connection);
+
+        V1_0::PathPartTable::PrepareForPackaging(connection);
+
+        V1_0::ManifestTable::PrepareForPackaging(connection, {
+            V1_0::VersionTable::ValueName(),
+            V1_0::ChannelTable::ValueName(),
+            V1_0::PathPartTable::ValueName(),
+            });
+
+        V1_0::TagsTable::PrepareForPackaging(connection);
+        V1_0::CommandsTable::PrepareForPackaging(connection);
+        PackageFamilyNameTable::PrepareForPackaging(connection, true);
+        ProductCodeTable::PrepareForPackaging(connection, true);
+
+        savepoint.Commit();
+
+        // Force the database to actually shrink the file size.
+        // This *must* be done outside of an active transaction.
+        SQLite::Builder::StatementBuilder builder;
+        builder.Vacuum();
+        builder.Execute(connection);
+    }
+
+    ISQLiteIndex::SearchResult Interface::Search(SQLite::Connection& connection, const SearchRequest& request)
+    {
+        // Update any system reference strings to be folded
+        SearchRequest foldedRequest = request;
+
+        auto foldIfNeeded = [](ApplicationMatchFilter& filter)
+        {
+            if ((filter.Field == ApplicationMatchField::PackageFamilyName || filter.Field == ApplicationMatchField::ProductCode) &&
+                filter.Type == MatchType::Exact)

```

Comment:
>filter.Type == MatchType::Exact [](start = 16, length = 31)

Should we just fold regardless of Matchtype for these 2 fields?

Reply:
The other MatchTypes use like for comparison, and so get folded as part of that.  While it wouldn't hurt anything, it isn't necessary.

---
In reply to: [488313354](https://github.com/microsoft/winget-cli/pull/564#discussion_r488313354) [](ancestors = 488313354)